### PR TITLE
DOCSP-11249

### PR DESCRIPTION
**Pull request by nathan-contino-mongo**
## Pull Request Info
- added example of executing a transaction with the Functions MQL connection
- removed tabs from the "setup" include for MongoDB CRUD Operations (only one tab just seemed silly)
- removed "transactions" limitation from service-limitations

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-11249

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/realm/pegasus/DOCSP-11249/mongodb/transactions.html

### CODE REVIEW

File: source/mongodb/transactions.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+The code snippets on this page demonstrate how to execute writes across
+multiple MongoDB queries in a single transaction that will "roll back"
+changes in the event of an error.
```
Comment (ccho-mongodb) [inline source/mongodb/transactions.txt:18]: Question: would it be useful to identify what a transaction is in the overview? Or is it unnecessary for the targeted personas?

File: source/mongodb/crud-snippets/transactions/transactions.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  try {
+    await session.withTransaction(async () => {
+
+      // Important:: You must pass the session to the operations
+      await accounts.updateOne({ name: userSubtractPoints }, { $inc: { browniePoints: (-1 * pointVolume) }}, { session });
+      await accounts.updateOne({ name: userAddPoints }, { $inc: { browniePoints: pointVolume }}, { session });
+      await browniePointsTrades.insertOne({ userAddPoints: userAddPoints, userSubtractPoints: userSubtractPoints, pointVolume: pointVolume }, { session });
+    }, transactionOptions);
+  } catch (err) {
+    await session.abortTransaction();
```
Comment (ccho-mongodb) [inline source/mongodb/crud-snippets/transactions/transactions.js:41]: Question: would it be useful to use comments to identify step 4 that you described in the ordered list under "to perform a transaction"?

File: source/mongodb/transactions.txt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   this object to the method called in the next step.
+
+#. Use the ``session.withTransaction()`` method to start a transaction.
+
+#. Execute the MongoDB queries that you would like to group together in
+   your transaction in the callback for this function, which needs to
+   either be ``async`` or return a ``Promise``. If the callback
+   encounters an error, you should abort the transaction with
+   ``session.abortTransaction()``, which you can handle in a ``Promise``
+   rejection handler with ``.catch()`` or a ``try/catch``.
```
Comment (ccho-mongodb) [inline source/mongodb/transactions.txt:39]: Question: I haven't seen the dot preceding the method (e.g. ".catch()"), was that intentional?

File: source/mongodb/transactions.txt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+To perform a transaction:
+
+1. Obtain and start a client session with ``client.startSession()``.
+
+#. (Optional) Define read and write settings for the transaction. Pass
+   this object to the method called in the next step.
+
+#. Use the ``session.withTransaction()`` method to start a transaction.
+
+#. Execute the MongoDB queries that you would like to group together in
```
Comment (ccho-mongodb) [inline source/mongodb/transactions.txt:34]: Nitpick: I would use "include" instead of "group together" since when I see the latter in the context of operations, I'm expecting to read about a "group" or even "union" operation.

File: source/mongodb/transactions.txt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+Example
+-------
+
+To perform a transaction:
+
+1. Obtain and start a client session with ``client.startSession()``.
+
+#. (Optional) Define read and write settings for the transaction. Pass
+   this object to the method called in the next step.
```
Comment (ccho-mongodb) [inline source/mongodb/transactions.txt:30]: Nitpick: I think you could mention the equivalency between settings and object in the first sentence to make it more clear. E.g. "Define a read and write settings object for the transaction."

File: source/mongodb/crud-snippets/transactions/transactions.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+exports = function(arg){
```
Comment (ccho-mongodb) [inline source/mongodb/crud-snippets/transactions/transactions.js:1]: Question: is it necessary to pass an argument to the function? I wasn't able to find where you used it.

File: source/mongodb/transactions.txt (hunk 3 above)
Comment (nathan-contino-mongo) [reply source/mongodb/transactions.txt:39]: It is intentional in this case to distinguish between the Promise handling catch() method and the try/catch syntax. I'm going to reword this to make the distinction clearer.

File: source/mongodb/crud-snippets/transactions/transactions.js (hunk 6 above)
Comment (nathan-contino-mongo) [reply source/mongodb/crud-snippets/transactions/transactions.js:1]: It is not -- I just never changed the declaration realm creates for you for functions. Fixed.

File: source/mongodb/crud-snippets/transactions/transactions.js (hunk 2 above)
Comment (nathan-contino-mongo) [reply source/mongodb/crud-snippets/transactions/transactions.js:41]: Going to put step 4 in the comments, but might put an extra step 4 (error handling) or step 5 to demonstrate that bit in a clearer way.
